tree/dataframe/src/RLoopManager.cxx

```diff
@@ -538,7 +547,16 @@ void RLoopManager::RunTreeProcessorMT()
       auto count = entryCount.fetch_add(nEntries);
       try {
          // recursive call to check filters and conditionally execute actions
-         while (r.Next()) {
+         auto validRead = [](TTreeReader &treereader) {
+            treereader.Next();
+            switch (treereader.GetEntryStatus()) {
+            case TTreeReader::kEntryValid: return true;
+            case TTreeReader::kIndexedFriendNoMatch: return true;
+            case TTreeReader::kMissingBranchWhenSwitchingTree: return true;
```

Comment:
It must return `true` in those three cases because the processing of the TTree still has to go on. Those three read statuses are "valid but with consequences that must be dealt with" and the rest of the code deals with them. As opposed to other read statuses of TTreeReader which actually would abort execution.

I agree this can be refactored in a separate function in an unnamed namespace